src/components/features/actionsMenu/MarkersTableActionsMenu.tsx

```diff
@@ -0,0 +1,135 @@
+import { getWarehouseManagementApi } from '@/services/generated-api'
+
+import { SubmitHandler } from 'react-hook-form'
+import { NewMarkerFormData, newMarkerSchema } from '@/schemas/newMarkerSchema'
+import FormDialog from '@/components/shared/FormDialog'
+import NewMarkerForm from '@/components/features/forms/NewMarkerForm'
+import { BodyType } from '@/services/api'
+import { MarkerFormDto } from '@/services/model'
+
+interface MarkersTableActionsMenuProps {
+  id: number
+  name: string
+}
+
+export default function MarkersTableActionsMenu({ id, name }: MarkersTableActionsMenuProps) {
+  const { t: translate } = useTranslation()
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const open = Boolean(anchorEl)
+  const [selectedOption, setSelectedOption] = React.useState<string | null>(null)
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleClose = () => {
+    setSelectedOption(null)
+    setAnchorEl(null)
+  }
+
+  const onDiscardClick = () => {
+    handleClose()
+  }
+
+  const actionHandler = (option: string) => {
+    setSelectedOption(option)
+  }
+
+  const queryClient = useQueryClient()
+
+  const mutationDelete = useMutation({
+    mutationFn: (id: number) => getWarehouseManagementApi().deleteApiMarkerDeleteId(id),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['markers'] })
+    },
+    onError: (error) => {
+      console.error('Грешка при заявката', error)
+    }
+  })
+
+  const onConfirmClick = () => {
+    mutationDelete.mutate(id)
+    handleClose()
+  }
+
+  const mutationUpdate = useMutation({
+    mutationFn: ({ id, data }: { id: number; data: BodyType<MarkerFormDto> }) =>
+      getWarehouseManagementApi().putApiMarkerEditId(id, data),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['markers'] })
+    },
+    onError: (error) => {
+      console.error('Грешка при заявката', error)
+    }
+  })
+
+  const handleSubmit: SubmitHandler<NewMarkerFormData> = (data) => {
+    console.log()
+    mutationUpdate.mutate({ id, data: { name: data.markerName } })
+  }
+
+  const options = [translate('actionsMenu.options.edit'), translate('actionsMenu.options.delete')]
+
+  return (
+    <div>
+      <IconButton
+        aria-label="more"
+        id="long-button"
+        aria-controls={open ? 'long-menu' : undefined}
+        aria-expanded={open ? 'true' : undefined}
+        aria-haspopup="true"
+        onClick={handleClick}>
+        <MoreHorizIcon />
+      </IconButton>
+      <Menu
+        id="long-menu"
+        MenuListProps={{
+          'aria-labelledby': 'long-button'
+        }}
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}>
+        {options.map((option) => (
+          <MenuItem key={option} onClick={() => actionHandler(option)}>
+            {option}
+          </MenuItem>
+        ))}
+      </Menu>
```

Comment:
use `TableActionsMenu` here, you just need to pass what items you want to render for the lsit and thats it. we dont want to write this for each actions menu

Reply:
Fixed